addons/mass_mailing/static/src/js/mass_mailing_sms_emojis.js

```diff
@@ -0,0 +1,14 @@
+odoo.define('mass_mailing.sms_emojis', function (require) {
+    "use strict";
+
+    var fieldRegistry = require('web.field_registry');
+    var MailEmojisMixin = require('mail.emoji_mixin');
+    var SmsWidget = require('sms.sms_widget');
+    var FieldEmojiCommon = require('mail.text_emojis_field').FieldEmojiCommon;
+    
+    // Add functionalities from emojis to sms widget to get a new one (SMS x EMOJIS)
+    var SmsEmojisWidget = SmsWidget.extend(MailEmojisMixin, FieldEmojiCommon);
+
+    fieldRegistry.add('sms_emojis', SmsEmojisWidget);
+    return SmsEmojisWidget;
+});
```

Comment:
Why do you need all this?

Reply:
Actually, the textField has already an SMS widget so to add the emojis we need to extend SMS widget with emojis behavior since we can't concat 2 widgets in the same field

Reply:
Ok I see it now.

After discussing with Thibault, here is what we are going to do:
- Add support for emojis in the base SmsWidget (in the sms module)
- The support is conditioned by a parameter on the widget "enable_emojis" that is specified in the field attribute: "options" (you should find good examples easily)
- If the parameter is not specified, the emojis are disabled by default

The changes to support the emojis with this new parameter will have to be split in a separate commit.

Then, you can enable the emojis_support on the sms field on the mailing.mailing form (in your main commit).

We will probably have to test that emojis are correctly supported in SMS by our providers, but we will see about that later.

cc @tde-banana-odoo

Reply:
My comment here above seems not addressed?
Did you miss it?